packages/financial-templates-lib/src/helpers/allowances.ts

```diff
@@ -17,25 +17,42 @@ export const setAllowance = async (
   currencyAddress: string
 ): Promise<
   | {
-      tx: TransactionReceipt;
-      spenderAddress: string;
-      currencyAddress: string;
-    }
+    tx: TransactionReceipt;
+    spenderAddress: string;
+    currencyAddress: string;
+  }
   | undefined
 > => {
   const { toBN } = web3.utils;
   const collateralToken = (new web3.eth.Contract(
     getAbi("ExpandedERC20"),
     currencyAddress
   ) as unknown) as ExpandedERC20Web3;
+  if (process.env.KMS_SIGNER) {
+    ownerAddress = process.env.KMS_SIGNER_ADDRESS!;
```

Comment:
suggestion for later -> when the bots start, we make sure `process.env.KMS_SIGNER_ADDRESS` is a valid ethereum address, and sanitize it for use. We use this sanitized and verified value, throughout the code.